apps/api/src/app/inbox/usecases/session/session.usecase.ts

```diff
@@ -0,0 +1,94 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { EnvironmentRepository } from '@novu/dal';
+import { ChannelTypeEnum, InAppProviderIdEnum, MessagesStatusEnum } from '@novu/shared';
+import {
+  AnalyticsService,
+  LogDecorator,
+  CreateSubscriber,
+  CreateSubscriberCommand,
+  SelectIntegrationCommand,
+  SelectIntegration,
+  AuthService,
+} from '@novu/application-generic';
+
+import { ApiException } from '../../../shared/exceptions/api.exception';
+import { SessionCommand } from './session.command';
+import { SessionResponseDto } from '../../dtos/session-response.dto';
+import { AnalyticsEventsEnum } from '../../utils';
+import { validateHmacEncryption } from '../../utils/encryption';
+import { NotificationsCount } from '../notifications-count/notifications-count.usecase';
+import { NotificationsCountCommand } from '../notifications-count/notifications-count.command';
+
+@Injectable()
+export class Session {
+  constructor(
+    private environmentRepository: EnvironmentRepository,
+    private createSubscriber: CreateSubscriber,
+    private authService: AuthService,
+    private selectIntegration: SelectIntegration,
+    private analyticsService: AnalyticsService,
+    private notificationsCount: NotificationsCount
+  ) {}
+
+  @LogDecorator()
+  async execute(command: SessionCommand): Promise<SessionResponseDto> {
+    const environment = await this.environmentRepository.findEnvironmentByIdentifier(command.applicationIdentifier);
+
+    if (!environment) {
+      throw new ApiException('Please provide a valid application identifier');
+    }
+
+    const inAppIntegration = await this.selectIntegration.execute(
+      SelectIntegrationCommand.create({
+        environmentId: environment._id,
+        organizationId: environment._organizationId,
+        userId: command.subscriberId,
```

Comment:
@LetItRock we need to remove the userId from here, as it's causing issues with Cache imploding. A fix was already pushed to next for the existing session initialize endpoint